### ISSUE TITLE
Tables not being marked as such

### ISSUE THREAD

**Issue by ChrisCRCB** [DavBfr/dart_pdf#1964]
**Describe the bug**
When creating tables, the resulting PDF doesn't seem to have them marked as tables. I'm navigating the resulting PDF with a screen reader, and according to the DOM, there are no tables in the document.

**To Reproduce**
Code snippet to reproduce the behavior:
```dart
import 'dart:io';

import 'package:pdf/widgets.dart' as pw;

Future<void> main(List<String> arguments) async {
  final document = pw.Document()
    ..addPage(
      pw.Page(
        build: (context) {
          final days = [
            ('Monday', 1),
            ('Tuesday', 2),
            ('Wednesday', 3),
            ('Thursday', 4),
            ('Friday', 5),
            ('Saturday', 6),
            ('Sunday', 7),
          ];
          return pw.TableHelper.fromTextArray(
            data: [
              ['Day Name', 'Day Number'],
              ...days.map((entry) {
                final (dayName, dayNumber) = entry;
                return [dayName, dayNumber];
              }),
            ],
          );
        },
      ),
    );
  final bytes = await document.save();
  File('poc.pdf').writeAsBytesSync(bytes);
}
```

**Expected behavior**
I should be able to use screen reader commands to jump to tables and between rows and columns. Apparently, the resulting table is not one. It's not such an issue with a simple table, like the one I use above, but the application I am working on generates invoices with a 5-column table, and it quickly becomes unreadable.

**Flutter Doctor**

```
[Γ£ô] Flutter (Channel stable, 3.47.1, on Microsoft Windows [Version 10.0.26200.9168], locale en-GB) [418ms]
    ΓÇó Flutter version 3.47.1 on channel stable at C:\Users\chris\flutter
    ΓÇó Upstream repository git@github.com:flutter/flutter.git
    ΓÇó Framework revision 6655482ec0 (6 days ago), 2026-08-19 10:07:23 -0700
    ΓÇó Engine revision 5d53178869
    ΓÇó Dart version 3.13.1
    ΓÇó DevTools version 2.60.0
    ΓÇó Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop, enable-windows-desktop, enable-android, enable-ios, cli-animations, enable-native-assets, enable-record-use, enable-swift-package-manager, omit-legacy-version-file, enable-lldb-debugging, enable-uiscene-migration

[Γ£ô] Windows Version (11 Home 64-bit, 25H2, 2009) [3.9s]

[!] Android toolchain - develop for Android devices (Android SDK version 36.0.0) [2.5s]
    ΓÇó Android SDK at C:\Users\chris\AppData\Local\Android\sdk
    ΓÇó Emulator version 37.1.11.0 (build_id 15917651) (CL:N/A)
    ΓÇó Platform android-36, build-tools 36.0.0
    ! Multiple adb binaries found. This can cause conflicts and device detection issues:
        - C:\Users\chris\AppData\Local\Android\Sdk\platform-tools\adb.exe
        - C:\Program Files\platform-tools\adb.exe
    ΓÇó Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    ΓÇó Java version OpenJDK Runtime Environment (build 25.0.2+-15348964-b329.117)
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses

[Γ£ô] Chrome - develop for the web [118ms]
    ΓÇó Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[Γ£ô] Visual Studio - develop Windows apps (Visual Studio Community 2026 18.8.3) [116ms]
    ΓÇó Visual Studio at C:\Program Files\Microsoft Visual Studio\18\Community
    ΓÇó Visual Studio Community 2026 version 18.8.12105.206
    ΓÇó Windows 10 SDK version 10.0.26100.0

[Γ£ô] Connected device (3 available) [626ms]
    ΓÇó Windows (desktop) ΓÇó windows ΓÇó windows-x64    ΓÇó Microsoft Windows [Version 10.0.26200.9168]
    ΓÇó Chrome (web)      ΓÇó chrome  ΓÇó web-javascript ΓÇó Google Chrome 151.0.7922.174
    ΓÇó Edge (web)        ΓÇó edge    ΓÇó web-javascript ΓÇó Microsoft Edge 151.0.4129.107

[Γ£ô] Network resources [622ms]
    ΓÇó All expected network resources are available.

! Doctor found issues in 1 category.
```

**Desktop (please complete the following information):**
 - [ ] iOS
 - [ ] Android
 - [ ] Browser
 - [✓] Windows
 - [ ] Linux
